### DISCUSSION TITLE
[CLOSED] [REQUEST/QUESTION] Showing a message without to grab X input

### DISCUSSION THREAD

**Discussion by renanwp2** [davatorium/rofi#1379] [Ideas/Feature Requests]
## Before creating a feature request:

- [x] I checked the *next* branch to see if the feature has already been
  implemented

- [x] I searched existing reports to see if it is already requested.

## What is the user problem or growth opportunity you want to see solved?

I want that showing a pop up message with/without icons, but without grabbing X input, i.e., without the user needing to press any keyboard input or any mouse button to close it. With this at hand, It would be possible to make an unified theme based on a Rofi theme, for example, it would be possible to have a menu launcher, sound indicator, notifications and much more with just Rofi installed, and yet matching user theme. In my case, I was looking for something that pop up in my screen wherever I want in a Rofi fashion. My dream as a /r/unixporn user. =P

## How do you know that this problem exists today?

Everytime I have tried to execute Rofi, it captured my keyboard input. I have tried to read multiple topics in Reddit and in the manual, but nothing a clear solution for this problem.

## Who will benefit from it? Why is this important?

There is yet an [user that asked something similar](https://www.reddit.com/r/unixporn/comments/av7kcv/rofi_notification_daemon/
), but everyone could benefit of it, I think, since it could be an user-made alternative to notify-send and yet a status monitor for something. Allowing to pop up a message without grabbing X input is desirable because we could send messages similarly to `notify-send`, but in `rofi -e "Message"` fashion with all its capabilities. The expect behavior would be to kill the Rofi window after a given number of seconds (or letting a specific key to kill it, while the others keys are not being captured). Is it already implemented? Or still not implemented?

## Version

Version: 1.6.1

## Configuration

Output of [rofi -help](https://gist.github.com/renanwp2/c1c47a3f6d347bc6f0a4b364bc3a5f5d/)

**Comment by DaveDavenport** [comment]
No and I don't directly  have a plan to implement a way to launch rofi in a modal form without grabbing the keyboard/mouse.

You can (in 1.7.0 the next release) have rofi close on no input after X seconds, see the (WIP) release notes.

If you need to show a notification to the user why not use notify-send? it seems to be the tool you are looking for.

(after replying I see it was edited with more info)

**Reply by renanwp2** [reply]
Sorry about letting my request without the proper format at the time you have read. It's a dream of a "/r/unixporn user". It would be possible to make an unified theme based solely on Rofi theme, for example, it would be possible to have a menu launcher, sound indicator, notifications and much more with just Rofi installed, and yet matching correctly my theme. One of my dreams. Hahaha

**Reply by DaveDavenport** [reply]
If people want this feature, please add a :+1:  to the original post.

**Reply by DaveDavenport** [reply]
p.s.  I solved it this way: https://github.com/DaveDavenport/rofication.   this is not very polished or finished.

**Reply by renanwp2** [reply]
I will try your solution. =)

**Reply by renanwp2** [reply]
I have tested and it's not what I have in mind. I was looking for something that pop up in my screen wherever I want in a Rofi fashion. For example, I am trying to make a status monitor for the volume always I turn up the volume of something playing. It could be used as a stylish user-made pop up for "notifications".

**Comment by renanwp2** [comment]
People, please, press the like button of this post. There is a chance of it getting implemented, if it turns to be relevant. Could you imagine a xclock made using Rofi? Could you imagine how it would turn /r/unixporn wonderful and how good it would be "robbing" others people config files? hahaha

**Reply by DaveDavenport** [reply]
keep in mind rofi is not a general purpose toolkit, its a launcher and list filter.

**Reply by renanwp2** [reply]
It's not needed, as a first step, to make it integrated with Dbus. The OS could try to use the usual notify-send, if needed. =) It eventually could turn your job easier, I suppose. It does not need to be thought not as a replacement for notify-send, but as a user-made popup without interrupting him, for example, in games or while the user is browsing.

**Reply by DaveDavenport** [reply]
I don't see how adding more features to maintain (esp. features I don't use myself) make my life easier? rofi is pretty bloated as is.

**Reply by renanwp2** [reply]
> I don't see how adding more features to maintain (esp. features I don't use myself) make my life easier? rofi is pretty bloated as is.

That's a good point! It' better not to put such a feature in the mainline project. The better I could hope is that someone could implement it as a fork, and not to hope it will work forever.  Now I can understand better the project itself and why this is too good. Sorry for implicitly making such a huge demand, that is, to hope for maintenance of a ad hoc feature. Thank you for giving your precious time. 

Now, it's wait for my best hope.
